backend/LeetWars.Core/LeetWars.Core.BLL/Services/ChallengeService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using AutoMapper;
+using LeetWars.Core.BLL.Interfaces;
+using LeetWars.Core.Common.DTO.Challenge;
+using LeetWars.Core.Common.DTO.Filters;
+using LeetWars.Core.Common.DTO.Language;
+using LeetWars.Core.DAL.Context;
+using LeetWars.Core.DAL.Entities;
+using LeetWars.Core.DAL.Enums;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static Microsoft.EntityFrameworkCore.DbLoggerCategory;
+
+namespace LeetWars.Core.BLL.Services
+{
+    public class ChallengeService : BaseService, IChallengeService
+    {
+        public ChallengeService(LeetWarsCoreContext context, IMapper mapper) : base(context, mapper) { }
+
+        public async Task<ICollection<ChallengePreviewDto>> GetChallengesAsync(ChallengesFiltersDto filters)
+        {
+            var challenges = _context.Challenges
+                    .Include(challenge => challenge.Tags)
+                    .Include(challenge => challenge.Author)
+                    .Include(challenge => challenge.Versions)
+                        .ThenInclude(version => version.Language)
+                     .Include(challenge => challenge.Versions)
+                        .ThenInclude(version => version.Solutions)
+                    .AsQueryable();
+
+            if (!string.IsNullOrEmpty(filters.Title))
+            {
+                challenges = challenges.Where(p => p.Title.ToLower().Contains(filters.Title.ToLower()));
+            }
+
+            if (filters.ChallengeStatus.HasValue)
+            {
+                challenges = challenges.Where(challenge => challenge.Versions.Any() &&
+                    challenge.Versions.All(version => version.Status == filters.ChallengeStatus));
```

Comment:
We cannot use All here, as the status is for a specific language
So probably it would be better to use Any here

Reply:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/68563967/eb8ec6da-d025-4e2d-8826-ef7aa2fd3936)

then we will receive challenges with not only the requested status

Reply:
is it okey?